presto-main/src/main/java/com/facebook/presto/sql/planner/plan/Assignments.java

```diff
@@ -78,18 +78,16 @@ public static Assignments of(Symbol symbol1, Expression expression1, Symbol symb
     }
 
     private final Map<Symbol, Expression> assignments;
-    private final List<Symbol> outputs;
 
     @JsonCreator
     public Assignments(@JsonProperty("assignments") Map<Symbol, Expression> assignments)
     {
         this.assignments = ImmutableMap.copyOf(requireNonNull(assignments, "assignments is null"));
-        this.outputs = ImmutableList.copyOf(assignments.keySet());
     }
 
     public List<Symbol> getOutputs()
     {
-        return outputs;
+        return ImmutableList.copyOf(assignments.keySet());
```

Comment:
> Also, ImmutableList.copyOf will not perform another copying when passed an immutable keySet().

Cool

> Assignments is a Map per its declaration type. I wouldn't like to rely on what's written in the ctor as it can change independently.

That's also my thinking by that seems not the coding standard we should adhere to. At least that's how i understand what @kokosing tried to persuade me.